src/instructlab/configuration.py

```diff
@@ -1354,52 +1354,100 @@ def storage_dirs_exist() -> bool:
     return all(os.path.exists(dirpath) for dirpath in dirs_to_check)
 
 
+# get_profile_mappings reads the profiles from disk where necessary and returns a mapping of the following format
 # {"GPU Name": ({ "gpu_count": NUM_GPUS, "vram_and_config": {"vram": TOTAL_VRAM, "config": TRAIN_CONFIG}})...}
-PROFILE_MAPPINGS = {
-    "L40s": (
-        {"gpu_count": 1, "vram_and_config": {"vram": 80, "config": SINGLE_L40}},
-        {
-            "gpu_count": 4,
-            "vram_and_config": {"vram": 192, "config": FOUR_L_FORTY_GPU},
-        },
-        {
-            "gpu_count": 8,
-            "vram_and_config": {"vram": 384, "config": EIGHT_L_FORTY_GPU},
-        },
-    ),
-    "L4": (
-        {"gpu_count": 1, "vram_and_config": {"vram": 24, "config": SINGLE_L4}},
-        {
-            "gpu_count": 8,
-            "vram_and_config": {"vram": 192, "config": EIGHT_L_FOUR_GPU},
-        },
-    ),
-    "A100": (
-        {
-            "gpu_count": 2,
-            "vram_and_config": {"vram": 160, "config": TWO_GPU_TRAIN_AH},
-        },
-        {
-            "gpu_count": 4,
-            "vram_and_config": {"vram": 320, "config": FOUR_GPU_TRAIN_AH},
-        },
-        {
-            "gpu_count": 8,
-            "vram_and_config": {"vram": 640, "config": EIGHT_GPU_TRAIN_AH},
-        },
-    ),
-    "H100": (
-        {
-            "gpu_count": 2,
-            "vram_and_config": {"vram": 160, "config": TWO_GPU_TRAIN_AH},
-        },
-        {
-            "gpu_count": 4,
-            "vram_and_config": {"vram": 320, "config": FOUR_GPU_TRAIN_AH},
-        },
-        {
-            "gpu_count": 8,
-            "vram_and_config": {"vram": 640, "config": EIGHT_GPU_TRAIN_AH},
-        },
-    ),
-}
+def get_profile_mappings() -> dict[str, tuple[dict[str, object], ...]]:
+    # read profiles from disk
+    TRAIN_DIR_EXPECTED_FILES = {
+        "A100_H100_x8.yaml",
+        "A100_H100_x4.yaml",
+        "A100_H100_x2.yaml",
+        "L40_x8.yaml",
+        "L40_x4.yaml",
+        "L4_x8.yaml",
+    }
+    # read the train cfg from disk, and use what is here to build the mappings.
+    PROFILE_NAMES_AND_CFGS = {}
```

Comment:
Probably shouldn't be in all caps anymore

Reply:
changed!